oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBConverter.java

```diff
@@ -0,0 +1,265 @@
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Storage;
+import org.apache.skywalking.oap.server.core.storage.type.StorageDataComplexObject;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+import org.apache.skywalking.oap.server.storage.plugin.banyandb.util.ByteUtil;
+
+import java.util.List;
+
+public class BanyanDBConverter {
+    public static class StorageToStream implements Convert2Entity {
+        private final MetadataRegistry.Schema schema;
+        private final RowEntity rowEntity;
+
+        public StorageToStream(String modelName, RowEntity rowEntity) {
+            this.schema = MetadataRegistry.INSTANCE.findMetadata(modelName);
+            this.rowEntity = rowEntity;
+        }
+
+        @Override
+        public Object get(String fieldName) {
+            MetadataRegistry.ColumnSpec spec = schema.getSpec(fieldName);
+            if (double.class.equals(spec.getColumnClass())) {
+                return ByteUtil.bytes2Double(rowEntity.getTagValue(fieldName));
+            } else {
+                return rowEntity.getTagValue(fieldName);
+            }
+        }
+
+        @Override
+        public byte[] getBytes(String fieldName) {
+            return rowEntity.getTagValue(fieldName);
+        }
+    }
+
+    @Slf4j
+    @RequiredArgsConstructor
+    public static class StreamToStorage implements Convert2Storage<StreamWrite> {
+        private final MetadataRegistry.Schema schema;
+        private final StreamWrite streamWrite;
+
+        @Override
+        public void accept(String fieldName, Object fieldValue) {
+            MetadataRegistry.ColumnSpec columnSpec = this.schema.getSpec(fieldName);
+            if (columnSpec == null) {
+                throw new IllegalArgumentException("fail to find field[" + fieldName + "]");
+            }
+            try {
+                this.streamWrite.tag(fieldName, buildTag(fieldValue, columnSpec.getColumnClass()));
+            } catch (BanyanDBException ex) {
+                log.error("fail to add tag", ex);
+            }
+        }
+
+        @Override
+        public void accept(String fieldName, byte[] fieldValue) {
+            try {
+                this.streamWrite.tag(fieldName, TagAndValue.binaryTagValue(fieldValue));
+            } catch (BanyanDBException ex) {
+                log.error("fail to add tag", ex);
+            }
+        }
+
+        @Override
+        public void accept(String fieldName, List<String> fieldValue) {
+            for (final String tagKeyAndValue : fieldValue) {
+                if (StringUtil.isEmpty(tagKeyAndValue)) {
+                    continue;
+                }
+                int pos = tagKeyAndValue.indexOf("=");
+                if (pos == -1) {
+                    continue;
+                }
+                String key = tagKeyAndValue.substring(0, pos);
+                String value = tagKeyAndValue.substring(pos + 1);
+                this.accept(key, value);
+            }
```

Comment:
Tag key=value is one tag, you don't need to split them.

Reply:
Fixed. Stored as string array